### PR TITLE
Task/restructure npm cli

### PR DESCRIPTION
**Pull request by dnitsch**
#### 📲 What

A description of the change.

#### 🤔 Why
		
Why it's needed, background context.
		
#### 🛠 How
		
More in-depth discussion of the change or implementation.

#### 👀 Evidence
		
Screenshots / external resources / links / etc.
Link to documentation updated with changes impacted in the PR
		 
#### 🕵️ How to test

Notes for QA

#### ✅ Acceptance criteria Checklist

- [ ] Code peer reviewed?
- [ ] Documentation has been updated to reflect the changes?
- [ ] Passing all automated tests, including a successful deployment?
- [ ] Passing any exploratory testing?
- [ ] Rebased/merged with latest changes from development and re-tested?
- [ ] Meeting the Coding Standards?

### CODE REVIEW

File: packages/template-cli/package.json [hunk 1]
```diff
@@ -29,13 +29,16 @@
...
+    "@types/minimist": "^1.2.0",
+    "@types/yargs": "^15.0.4",
     "chalk": "^3.0.0",
     "fs-extra": "^8.1.0",
     "prompts": "^2.3.1",
     "replace-in-file": "^5.0.2",
     "simple-git": "^1.132.0",
     "simple-winston-logger-abstraction": "^2.0.0",
-    "typescript": "3.7.5"
+    "typescript": "3.7.5",
```
Comment (estherthetesteramido) [inline packages/template-cli/package.json]: We need this in dependenciess?

File: packages/template-cli/src/domain/prompt.ts [hunk 2]
```diff
@@ -18,31 +16,46 @@ let exitMessage: ExitMessage = <ExitMessage>{}
...
+    try {
+        await Utils.writeOutConfigFile(`sample.bootstrap-config.json`)
+        exitMessage.code = 0
+        exitMessage.message = 
+`Sample config file has been written out to current directory \n
+Please re-run the CLI with the following command. \n
+----- \n
+npx @amido-stacks/scaffolding-cli run -c sample.bootstrap-config.json \n
+----- \n
+NB: IF you haven't gone through the advanced setup - plese ensure you have replaced all the relevant values.
```
Comment (estherthetesteramido) [inline packages/template-cli/src/domain/prompt.ts]: Spelling `please`

File: packages/template-cli/src/__tests__/domain/workers/utils.test.ts [hunk 3]
```diff
@@ -32,20 +40,24 @@ let ssr_tfs_aks: Array<FolderMap> = [
...
 }
 
+// let mockGit = gitP(temp_dir)
+
 describe("utils class tests", () => {
     beforeEach(() => {
         mockCopy.mockClear()
         mockMove.mockClear()
         mockReplace.mockClear()
+        // mockClone.mockClear()
```
Comment (estherthetesteramido) [inline packages/template-cli/src/__tests__/domain/workers/utils.test.ts]: remove

File: packages/template-cli/src/__tests__/domain/workers/utils.test.ts [hunk 4]
```diff
@@ -103,10 +125,9 @@ describe("utils class tests", () => {
         })
         it("copyWorker should return a code of ENOENT when error occurs", async () => {
             mockCopy.mockImplementationOnce(() => {
-                throw {code: "ENOENT", message: new Error("Something weird happened")}
+                throw { code: "ENOENT", message: new Error("Something weird happened") }
```
Comment (estherthetesteramido) [inline packages/template-cli/src/__tests__/domain/workers/utils.test.ts]: Better error?

File: packages/template-cli/src/domain/prompt.ts [hunk 5]
```diff
@@ -84,25 +97,35 @@ async function selectFlow(selection: CliAnswerModel): Promise<ExitMessage> {
...
+//             // keeping it simple for now
+//             // writing out sample config without value replacement for now
+//             await Utils.writeOutConfigFile(resolve(__dirname, 'config/sample.bootstrap-config.json'), `${selection.project_name}.bootstrap-config.json`)
+//             console.log(chalk.yellowBright(
+// `Sample config file has been written out to current directory \n
+// Please re-run the CLI with the following command. \n
+// ----- \n
+// npx @amido-stacks/scaffolding-cli ${selection.project_name}.bootstrap-config.json \n
+// ----- \n
+// NB: IF you haven't gone through the advanced setup - plese ensure you have replaced all the relevant values.
```
Comment (estherthetesteramido) [inline packages/template-cli/src/domain/prompt.ts]: Spelling

File: packages/template-cli/src/index.ts [hunk 6]
```diff
@@ -1,19 +1,89 @@
...
-// main cli entry
-(async () => {
-    const default_project_name = basename(resolve(process.cwd()))
-    const args = process.argv
+
+async function cliCommand(argv: CliOptions) {
     try {
-        const response: ExitMessage = await runCli(default_project_name, args.slice(2))
+        const default_project_name = basename(resolve(process.cwd()));
+        // let argv = giveMeArgs()
```
Comment (estherthetesteramido) [inline packages/template-cli/src/index.ts]: Remove?

File: packages/template-cli/src/index.ts [hunk 7]
```diff
@@ -1,19 +1,89 @@
...
+        } else {
+            console.log(chalk.cyan(yargs.showHelp())) //"Please select an appropriate option"
+            return process.exit(0)
+        }
         console.log(chalk.cyan(response.message))
         return process.exit(0)
     } catch (ex) {
-        console.log(ex.message)
+        let exCaught = ex as ExitMessage
+        console.log(chalk.red(exCaught.code))
```
Comment (estherthetesteramido) [inline packages/template-cli/src/index.ts]: One log?

File: packages/template-cli/src/index.ts [hunk 8]
```diff
@@ -1,19 +1,89 @@
...
+    .usage('Usage: npx @amidostacks/scaffolding-cli <command> [options]')
+    .example('scaffolding-cli run -gsc', 'Dry run to only generate a sample config json')
+    .example('scaffolding-cli run -c sample.bootstrap.config.json', 'Run Scaffolding CLI with a options specified in a config file')
+    .example('scaffolding-cli run', 'Run Scaffolding CLI with interactive prompts')
+    .showHelpOnFail(true)
+    .completion()
+    // .showHelp()
+    .demandCommand()
+    .epilog("Thanks")
+    // .demandOption("", "Either -c or -i is required")
```
Comment (estherthetesteramido) [inline packages/template-cli/src/index.ts]: Remove?

File: packages/template-cli/src/index.ts (hunk 7 above)
Comment (dnitsch) [reply packages/template-cli/src/index.ts]: it's better to log out the message and code separately I think - as opposed to concat'ing them together no?

**Comment by dnitsch** [comment]
> Do you want to `lerna version` on this to bump the package?

yes - doing that now

**Pull request merged**